src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -166,6 +171,11 @@ bool QgsPointCloudLayerRenderer::render()
       QgsDebugMsgLevel( "canceled", 2 );
       break;
     }
+    if ( context.pointsRendered() > mRenderer->pointBudget() )
+    {
+      QgsDebugMsgLevel( "point budget exceeded", 2 );
+      break;
+    }
```

Comment:
I think for 2D rendering we do not need to worry about point budget at all, what do you think @nyalldawson ?

I believe the point budget is only important for 3D rendering because it is our way how to protect the GPU from getting overwhelmed by the draw requests from QGIS (which could even freeze the whole system). But with 2D rendering, even if we would render huge amounts of points, we are not going to exhaust the memory or anything like that - it will just take longer time...

Reply:
Yeah, let's skip it for now